File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1.Tests/Models/FakePrincipal.vb
Imports System.Security.Principal

Public Class FakePrincipal
    Implements IPrincipal

    Private _name As String

    Public Sub New(ByVal name As String)
        _name = name
    End Sub

#Region "IPrincipal Members"

    Public ReadOnly Property Identity() As IIdentity Implements IPrincipal.Identity
        Get
            Return New FakeIdentity(_name)
        End Get
    End Property

    Public Function IsInRole(ByVal role As String) As Boolean Implements IPrincipal.IsInRole
        Throw New NotImplementedException()
    End Function

#End Region
End Class

File: AspNetMvcUnleashed/0672329980_source/CommonCode/MvcFakes/MvcFakes/FakePrincipal.cs
using System.Security.Principal;
using System;
using MvcApplication1.Tests.Models;

public class FakePrincipal : IPrincipal
{

    private string _name;

    public FakePrincipal(string name)
    {
        _name = name;
    }

    #region IPrincipal Members

    public IIdentity Identity
    {
        get { return new FakeIdentity(_name); }
    }

    public bool IsInRole(string role)
    {
        throw new NotImplementedException();
    }

    #endregion
}
